### PR TITLE
🚑 Catch unset stations in distance calculation progress

### PR DESCRIPTION
**Pull request by HerrLevin**

### CODE REVIEW

**Comment by jeyemwey** [comment]
Whats the real diff here besides that you moved the methods?

File: app/Http/Controllers/Backend/GeoController.php [hunk 1]
```diff
@@ -159,6 +103,74 @@ public static function calculateDistanceBetweenCoordinates(
...
+     * @param HafasTrip $hafasTrip
+     *
+     * @return mixed
+     * @throws JsonException
+     */
+    private static function getPolylineWithTimestamps(HafasTrip $hafasTrip): stdClass {
+        $geoJsonObj = json_decode($hafasTrip->polyline->polyline, false, 512, JSON_THROW_ON_ERROR);
+        $stopovers  = $hafasTrip->stopoversNEW;
+
+        $stopovers = $stopovers->map(function($stopover) {
```
Comment (HerrLevin) [inline app/Http/Controllers/Backend/GeoController.php:149]: added this map here

File: app/Http/Controllers/Backend/GeoController.php [hunk 2]
```diff
@@ -159,6 +103,74 @@ public static function calculateDistanceBetweenCoordinates(
...
+        foreach ($geoJsonObj->features as $polylineFeature) {
+            if (!isset($polylineFeature->properties->id)) {
+                continue;
+            }
+
+            $stopover = $stopovers->where('trainStation.ibnr', $polylineFeature->properties->id)
+                                  ->where('passed', false)
+                                  ->first();
+
+            if (is_null($stopover)) {
```
Comment (HerrLevin) [inline app/Http/Controllers/Backend/GeoController.php:163]: added this catch here

**Review by HerrLevin — COMMENTED** [review]
sorry, phpstorm gone rogue.

File: app/Http/Controllers/Backend/GeoController.php (hunk 1 above)
Comment (MrKrisKrisu) [reply app/Http/Controllers/Backend/GeoController.php:149]: I don't see any relevanc for this map... you don't need `passed` to be set by default?

**Pull request merged**